helm/event-exporter-app/templates/configmap.yaml

```
@@ -17,6 +16,11 @@ data:
     route:
       routes:
 
+        # We want to post kubernetes events from cluster-operator and aws-operator to grafana.
+        - match:
+          - kind: "Cluster|AWSCluster|MachineDeployment|AWSMachineDeployment|G8sControlPlane|AWSControlPlane"
+            receiver: "grafana-firecracker"
+        {{- if .Values.Installation.V1.Secret.EventExporterApp.Slack }}
```

* Let's have more visual representation to help our eyes. I am old. 
* Checking for `Secret.EventExporterApp.Slack` will make the helm templating fail for installations for which we do not have the token configured. This is why the struct should be checked for in case we have nested data structures injected.  

```suggestion

        {{- if .Values.Installation.V1.Secret.EventExporterApp }}
```

---

Hm, but while `EventExporterApp` is defined on all installations, `EventExporterApp.Slack` is not. Because of that, shouldn't we check for that specifically?

---

Why do you think `Secret.EventExporterApp` is defined in all installations? I only ensured the slack tokens on installations we care about.

---

Meaning, just because `EventExporterApp` is present does not ensure that `Slack` is present anymore, which is why I want to check for that.

---

I got it now. With your PR we ensure the structure everywhere. Got it.

---

Sorry, I should have added a comment to explain this!